lib/SILOptimizer/Transforms/DeadCodeElimination.cpp

```diff
@@ -258,6 +258,19 @@ void DCE::markLive(SILFunction &F) {
         }
         continue;
       }
+      if (isa<EndAccessInst>(&I)) {
+        if (auto *access = dyn_cast<BeginAccessInst>(I.getOperand(0))) {
+          addReverseDependency(access, &I);
+        } else {
+          markValueLive(&I);
+        }
+        continue;
+      }
+      if (auto access = dyn_cast<BeginAccessInst>(&I)) {
```

Comment:
begin_access is marked as having side effects so, I still have to have this bit of logic.

Reply:
That logic looks wrong. It should be sufficient to exclude begin_access in seemsUseful().
If there is a use (other than end_access), the begin_access will be marked as life in propagateLiveness().